### ISSUE TITLE
validator depositdata

### ISSUE THREAD

**Issue by MysticRyuujin** [wealdtech/ethdo#18]
I am completely lost with regards to the validator depositdata functions now.

It used to be that I could do:
```
ethdo validator depositdata --validatoraccount="Validators/1" --withdrawalaccount="Withdrawal/Withdrawal" --depositvalue="32Ether" --passphrase="secret"
```

I could then feed that output to `ethereal`

It doesn't look like that's the proper course of action anymore. As I just get:
```
ethdo validator depositdata --validatoraccount="Validators/1"
Failed to obtain validator account
```

I know everything is in flux and keeps changing 😞 

To make matters worse, I must have used an older version to generate my deposits, because they're invalid, even though I used ethereal to submit them. Them being invalid made me download the latest `ethdo` and try to re-generate the deposit data when I saw this.

This is confusing since I haven't seen any mention to the change in the signature scheme or whatever would cause my data to be invalid?

Here's a sample:

https://beaconscan.com/validator/0xa49d4d036e7278ba6190ee1cadc2b8f75f31f5b83436d023f1cd7e028f312f065c8b378f6b78db3a0463aeb41fe31253#deposits

**Comment by mcdee** [comment]
For the error, I would assume it means that account doesn't exist.  What happens if you try `ethdo account info --account=Validators/1`?

Regarding the invalid deposit: part of the signature includes what is known as the "fork version".  The fork version is chain-specific, so if you were running (for example) an Onyx node and try to create a deposit it will be specific to Onyx.

So to create deposit data you have two options.  One is to connect to a node for the chain for which you are generating the data, the other is to supply the fork version manually.

**Comment by MysticRyuujin** [comment]
Running the command:
```
ethdo account info --account=Validators/1
Public key: 0x9046965231b80fb4139e1e04504409676e2b0684403d6e538fcfb0c22039f8d7278ebbc189ae25751c19dccbf0bf318a
Path: m/12381/3600/0/0`
```

When I ran the command with `1.5.1` I got the error about not being able to connect to a node, which is why I asked. When I run the command with `1.5.4` I get the `Failed to obtain validator account` error. So I thought maybe that this is just a case of misleading error message?

I recently modified my docker setup so that it didn't expose port 4000 on my host, which is where I run `ethdo`, so I thought maybe that was the problem, but I just switched back to exposing 4000 and the command still isn't working...

So to summarize, with `1.5.4`

The validator account exists, I can get the private key with the `ethdo account key` so I know my passphrase is correct, and my Medalla Prysm beacon node is listening on 0.0.0.0:4000 so it should be able to talk to it.

```
PS > ethdo account info --account=Validators/1
Public key: 0x9046965231b80fb4139e1e04504409676e2b0684403d6e538fcfb0c22039f8d7278ebbc189ae25751c19dccbf0bf318a
Path: m/12381/3600/0/0

PS > ethdo validator depositdata --validatoraccount="Validators/1" --withdrawalaccount="Withdrawal/Withdrawal" --depositvalue="32Ether" --passphrase="secret"
Failed to obtain validator account

PS > docker logs --tail 100 prysm-beacon-node
time="2020-08-01 12:46:16" level=info msg="Checking DB" database-path="/data/beaconchaindata" prefix=node
time="2020-08-01 12:46:16" level=warning msg="No bootstrap addresses supplied" prefix=p2p
time="2020-08-01 12:46:18" level=info msg="Starting beacon node" prefix=node version="Prysm/v1.0.0-alpha.16/f414cf36b99dea2d997d0141acb3d1a319fb5578. Built at: 2020-07-31 02:55:30+00:00"
time="2020-08-01 12:46:18" level=info msg="RPC-API listening on port" address="0.0.0.0:4000" prefix=rpc
```

Just in case:
```
ethdo account info --account="Withdrawal/Withdrawal"
Public key: 0xaea0ea15666833737ee628d9beb0161fdc334608c543c498a85b4d6788de58a7903e5b4780722a0b5988eb20fdea21e5
Path: m/12381/3600/0/0
```

**Comment by mcdee** [comment]
Thank you for the detailed information.  A bug was introduced in v1.5.4 validator deposits that has been addressed in v1.5.5.  This version should now be available for download; if it fails to address this issue please open a new issue with the specifics of the problem.
